File: freak-fights-fan/shared/features/fights/helpers/fightHelpers.ts
import { CreateFight } from "../commands/CreateFight";

class FightHelpers {
    static haveUniqueFighters(teams: CreateFight.CreateTeamModel[]): boolean {
        const allFightersIds: number[] = [];

        for (const createTeamModel of teams) {
            for (const fighter of createTeamModel.fighters) {
                if (allFightersIds.includes(fighter.fighterId)) {
                    return false;
                }

                allFightersIds.push(fighter.fighterId);
            }
        }

        return true;
    }
}

File: FreakFightsFan.Shared/Features/Fights/Helpers/FightHelpers.cs
using static FreakFightsFan.Shared.Features.Fights.Commands.CreateFight;

namespace FreakFightsFan.Shared.Features.Fights.Helpers;

public static class FightHelpers
{
    public static bool HaveUniqueFighters(List<CreateTeamModel> teams)
    {
        var allFightersIds = new List<int>();

        foreach (var fighter in teams.SelectMany(createTeamModel => createTeamModel.Fighters))
        {
            if (allFightersIds.Contains(fighter.FighterId))
            {
                return false;
            }

            allFightersIds.Add(fighter.FighterId);
        }

        return true;
    }
}